gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemDelegationTokensTest.java

```diff
@@ -139,11 +146,99 @@ public void testDelegationTokenStatistics() throws IOException {
+        .isEqualTo(1);
+    assertThat(fs.getIOStatistics().counters().get(DELEGATION_TOKENS_ISSUED.getSymbol()))
+        .isEqualTo(1);
+
+    // second test for renewer which is null
+    Token<?> dt2 = fs.getDelegationToken(null);
+    assertThat(fs.getIOStatistics().counters().get(INVOCATION_GET_DELEGATION_TOKEN.getSymbol()))
+        .isEqualTo(2);
+    assertThat(fs.getIOStatistics().counters().get(DELEGATION_TOKENS_ISSUED.getSymbol()))
+        .isEqualTo(2);
+    fs.close();
+  }
+
+  @Test
+  public void testDelegationAreSharedForSharedTokensConfigAndSameRenewers() throws IOException {
+
+    GoogleHadoopFileSystem fs = new GoogleHadoopFileSystem();
+    fs.initialize(new Path("gs://test/").toUri(), loadConfig(GCSTokenInstantiationStrategy.SHARED));
+
+    GoogleHadoopFileSystem fsOther = new GoogleHadoopFileSystem();
+    fsOther.initialize(
+        new Path("gs://test2/").toUri(), loadConfig(GCSTokenInstantiationStrategy.SHARED));
+
+    Token<?> dt1 = fs.getDelegationToken("current_user");
+    Token<?> dt2 = fsOther.getDelegationToken("current_user");
+    assertThat(dt1).isEqualTo(dt2);
+
+    fs.close();
+    fsOther.close();
+  }
+
+  @Test
+  public void testDelegationAreSharedForSharedTokensConfigAndNullRenewers() throws IOException {
+
+    GoogleHadoopFileSystem fs = new GoogleHadoopFileSystem();
+    fs.initialize(new Path("gs://test/").toUri(), loadConfig(GCSTokenInstantiationStrategy.SHARED));
+
+    GoogleHadoopFileSystem fsOther = new GoogleHadoopFileSystem();
+    fsOther.initialize(
+        new Path("gs://test2/").toUri(), loadConfig(GCSTokenInstantiationStrategy.SHARED));
+
+    Token<?> dt1 = fs.getDelegationToken(null);
+    Token<?> dt2 = fsOther.getDelegationToken(null);
+    assertThat(dt1).isEqualTo(dt2);
+
+    fs.close();
+    fsOther.close();
+  }
+
+  @Test
+  public void testDelegationAreNotSharedForSharedTokensConfigAndDifferentRenewers()
+      throws IOException {
+
+    GoogleHadoopFileSystem fs = new GoogleHadoopFileSystem();
+    fs.initialize(new Path("gs://test/").toUri(), loadConfig(GCSTokenInstantiationStrategy.SHARED));
+
+    GoogleHadoopFileSystem fsOther = new GoogleHadoopFileSystem();
+    fsOther.initialize(
+        new Path("gs://test2/").toUri(), loadConfig(GCSTokenInstantiationStrategy.SHARED));
+
+    GoogleHadoopFileSystem fsNullRenewer = new GoogleHadoopFileSystem();
+    fsNullRenewer.initialize(
+        new Path("gs://test3/").toUri(), loadConfig(GCSTokenInstantiationStrategy.SHARED));
+
+    Token<?> dt1 = fs.getDelegationToken("current_user");
+    Token<?> dt2 = fsOther.getDelegationToken("current_user2");
+    Token<?> dt3 = fsOther.getDelegationToken(null);
+
+    assertThat(dt1).isNotEqualTo(dt2);
+    assertThat(dt1).isNotEqualTo(dt3);
+    assertThat(dt3).isNotEqualTo(dt2);
+
+    fs.close();
+    fsOther.close();
+    fsNullRenewer.close();
+  }
+
+  private Configuration loadConfig(GCSTokenInstantiationStrategy gcsTokenInstantiationStrategy) {
```

Comment:
I think you can overload this method definition so default instantiation strategy is one instance per service.